src/nvim/os/env.c

```diff
@@ -423,43 +423,41 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(const char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
-  int vimruntime;
-
-
-  p = (char_u *)os_getenv((char *)name);
-  if (p != NULL && *p == NUL)       /* empty is the same as not set */
-    p = NULL;
-
-  if (p != NULL) {
-    return p;
+  const char *kos_env_path = os_getenv(name);
```

Comment:
The style guide actually says this of "All compile-time constants..."

http://neovim.org/develop/style-guide.xml?showone=Constant_Names#Constant_Names